addons/web/static/src/views/view_button/view_button.js

```diff
@@ -46,10 +49,23 @@ export class ViewButton extends Component {
     }
 
     onClick() {
-        this.env.onClickViewButton({
-            clickParams: this.props.clickParams,
-            record: this.props.record,
-        });
+        if (this.props.confirm) {
+            const dialogProps = {
+                body: this.props.confirm,
+                confirm: () =>
+                    this.env.onClickViewButton({
```

Comment:
all this within onClickViewButton ?
+
make sure the dialog is closed when the view is